### PR TITLE
Add dynamic config for sdk worker options

### PR DESCRIPTION
**Pull request by dnr**
**What changed?**
- Add process-level dynamic config for sdk worker sticky cache size.
- Remove direct calls to `worker.New` and use factory.
- Collapse sdk client and worker factories to one, since they're always used together.
- Add dynamic config for per-namespace workers.

Use like:
```
worker.perNamespaceWorkerOptions:
- value:
    MaxConcurrentWorkflowTaskPollers: 10
    WorkerLocalActivitiesPerSecond: 100
```

**Why?**
Allows more sdk worker tuning.

**How did you test it?**
not yet

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/worker/service.go [hunk 1]
```diff
@@ -114,6 +113,7 @@ type (
 		BatcherConcurrency                    dynamicconfig.IntPropertyFnWithNamespaceFilter
 		EnableParentClosePolicyWorker         dynamicconfig.BoolPropertyFn
 		PerNamespaceWorkerCount               dynamicconfig.IntPropertyFnWithNamespaceFilter
+		PerNamespaceWorkerOptions             dynamicconfig.MapPropertyFnWithNamespaceFilter
```
Comment (Quinn-With-Two-Ns) [inline service/worker/service.go:116]: Wouldn't we want these configs to also be per worker, the schedule worker would probably have different constraints then the batch worker?

File: common/dynamicconfig/constants.go [hunk 2]
```diff
@@ -713,6 +713,10 @@ const (
 	WorkerParentCloseMaxConcurrentWorkflowTaskPollers = "worker.ParentCloseMaxConcurrentWorkflowTaskPollers"
 	// WorkerPerNamespaceWorkerCount controls number of per-ns (scheduler, batcher, etc.) workers to run per namespace
 	WorkerPerNamespaceWorkerCount = "worker.perNamespaceWorkerCount"
+	// WorkerPerNamespaceWorkerOptions are SDK worker options for per-namespace worker
+	WorkerPerNamespaceWorkerOptions = "worker.perNamespaceWorkerOptions"
 	// WorkerEnableScheduler controls whether to start the worker for scheduled workflows
 	WorkerEnableScheduler = "worker.enableScheduler"
+	// WorkerStickyCacheSize controls the sticky cache size for all SDK workers on the worker node
```
Comment (Quinn-With-Two-Ns) [inline common/dynamicconfig/constants.go:720]: Might be worth calling out that the sticky cache is shared between all workers in the processes

File: service/worker/service.go (hunk 1 above)
Comment (dnr) [reply service/worker/service.go:116]: We combine all the per-namespace workflows into one worker for efficiency. If it really becomes impossible to find settings that work for the combined worker we could reconsider, but for now we should try to tune them together.

**Review by Quinn-With-Two-Ns — APPROVED** [review]

File: service/worker/pernamespaceworker.go [hunk 3]
```diff
@@ -398,16 +432,26 @@ func (w *perNamespaceWorker) startWorker(
...
-	sdkoptions.MaxConcurrentWorkflowTaskPollers = 2 * multiplicity
-	sdkoptions.MaxConcurrentActivityTaskPollers = 2 * multiplicity
+	// increase these if we're supposed to run with more multiplicity
+	sdkoptions.MaxConcurrentWorkflowTaskPollers *= multiplicity
+	sdkoptions.MaxConcurrentActivityTaskPollers *= multiplicity
 	sdkoptions.OnFatalError = w.onFatalError
 
 	// this should not block because the client already has server capabilities
-	sdkworker := w.wm.sdkWorkerFactory.New(client, primitives.PerNSWorkerTaskQueue, sdkoptions)
+	sdkworker := w.wm.sdkClientFactory.NewWorker(client, primitives.PerNSWorkerTaskQueue, sdkoptions)
```
Comment (alexshtin) [inline service/worker/pernamespaceworker.go:454]: `sdkwrk`?

Comment (dnr) [reply service/worker/pernamespaceworker.go:454]: `worker` is unused in this scope. I'll just use that. (I think previously there was another variable/arg named `worker`)

**Review by alexshtin — APPROVED** [review]

**Pull request merged**